### PR TITLE
Restoring Append-only security requirements

### PR DESCRIPTION
**Pull request by fournet**
See issue  #199 

The requirements in ### Append-only Log are restored (and much simplified) from earlier drafts, notably what used to be in ### Append-only Log Security Requirements.

The other changes are minor changes for uniformity about the Append-only Log.

### CODE REVIEW

File: draft-ietf-scitt-architecture.md
Comment (OR13) [inline draft-ietf-scitt-architecture.md:375]: I wonder if the Log is allowed to accumulate anything OTHER than "Signed Statements"... for example certificates, or JWTs.

File: draft-ietf-scitt-architecture.md
Comment (OR13) [inline draft-ietf-scitt-architecture.md:426]: I can imagine readers might get hung up on "universally-verifiable proof of registration".

I wonder if we should define SCITT receipts as "proof of registration" that build on COSE receipts, which are signed inclusion proofs.

**Review by OR13 — APPROVED** [review]

**Comment by OR13** [comment]
Not sure why the CI is failing, but I don't think that should block merging PRs, anyone can fix this after merging by running:

```
make fix-lint
```

File: draft-ietf-scitt-architecture.md
Comment (SteveLasker) [reply draft-ietf-scitt-architecture.md:375]: I preferred the previous version as it incorporates generating receipts. This also changes registered to accumulates, which only adds more concepts.

Comment (ad-l) [reply draft-ietf-scitt-architecture.md:375]: Allowing the TS to use the log for audit information, in particular intermediate certificates or JWTs, is a pretty legitimate use case, so I would say yes

File: draft-ietf-scitt-architecture.md
Comment (ad-l) [inline draft-ietf-scitt-architecture.md:377]:
```suggestion
Optionally, the Ledger may also record other collateral information for Auditors, such as governance information, records of management operations or security incidents, intermediate certificates used for Issuer authentication, etc.

```

**Review by ad-l — APPROVED** [review]

File: draft-ietf-scitt-architecture.md
Comment (aj-stein-nist) [inline draft-ietf-scitt-architecture.md:226]: Isn't this the only one we have defined? Why not put them in a table and make it small changes to add if there is presumably more than one? It seems we have COMETRE and the CCF profile for COMETER, so I am not sure why we would want to emphasize this when no public draft or IETF spec drafts an alternative one.

Am I missing something?

File: draft-ietf-scitt-architecture.md
Comment (aj-stein-nist) [reply draft-ietf-scitt-architecture.md:375]: So we are saying, @ad-l we MUST add alternatives data to Signed Statements? SHOULD?

This has been discussed a few times in calls and on the mailing list, why MUST or SHOULD it be in the log and not something computed and returned by the API? I do not oppose that change, but why open the door?

And you want the certs and JWTs themselves in the log, @ad-l? I can understand signed or transparent statements if properly sealed, but this reads as using the log for generic storage. Beyond my personal opinion, I thought many in the group oppose this notion generally.

File: draft-ietf-scitt-architecture.md
Comment (aj-stein-nist) [reply draft-ietf-scitt-architecture.md:377]: If it is just a different kind of payload, why must we be explicit with this?

File: draft-ietf-scitt-architecture.md
Comment (JAG-UK) [reply draft-ietf-scitt-architecture.md:426]: I like 'offline verifiable' a lot more than 'universally'

File: draft-ietf-scitt-architecture.md
Comment (SteveLasker) [reply draft-ietf-scitt-architecture.md:378]: Various implementations will support many different scenarios. I'd suggest change to 375/376 should be reverted, and remove the additional 378

File: draft-ietf-scitt-architecture.md
Comment (SteveLasker) [inline draft-ietf-scitt-architecture.md:377]: Proposal to revert this change, as "accumulates", as we've moved away from the Transparency Service recording Signed Statement contgents, as that may introduce PII or other redaction scenarios that would be impossible. Rather the Transparency Service registers signed statements, producing receipts.
```suggestion
The Append-only Log is the verifiable data structure that records Signed Statements and supports the production of Receipts.
```

File: draft-ietf-scitt-architecture.md
Comment (SteveLasker) [inline draft-ietf-scitt-architecture.md:428]:
```suggestion
: there is no fork in the Append-only Log.
Everyone with access to its content sees the same collection of Signed Statements and can check that it is consistent with any Receipts they have verified.
```

File: draft-ietf-scitt-architecture.md
Comment (SteveLasker) [inline draft-ietf-scitt-architecture.md:424]:
```suggestion
: once included in the verifiable data structure, a Signed Statement cannot be modified, deleted, or reordered; hence its Receipt provides an offline verifiable proof of registration.
```

File: draft-ietf-scitt-architecture.md
Comment (SteveLasker) [reply draft-ietf-scitt-architecture.md]: > I like 'offline verifiable' a lot more than 'universally'

Suggested change added

File: draft-ietf-scitt-architecture.md
Comment (robinbryce) [reply draft-ietf-scitt-architecture.md]: We need to distinguish between the log and the verifiable data structure if we are doing this. A merkle based append only log is just a bunch of hashes. The signed statements are committed to it, not "in it". I strongly prefer the original wording.

File: draft-ietf-scitt-architecture.md
Comment (robinbryce) [reply draft-ietf-scitt-architecture.md]: This just isn't true tho is it ? If a service has a take down request or whatever, it is legaly obliged to destroy the statement. This is usually accomplished by deleting the pre-image or "forgeting" some encrypting key

File: draft-ietf-scitt-architecture.md
Comment (robinbryce) [reply draft-ietf-scitt-architecture.md:378]: The original wording here is much more preferable. The verifiabiliy depends on the ability to produce proofs (receipts) not where the signed statements are stored.

File: draft-ietf-scitt-architecture.md
Comment (robinbryce) [reply draft-ietf-scitt-architecture.md]:
```suggestion
once included in the verifiable data structure, a Signed Statement cannot be modified, deleted, or reordered without invalidating the cryptographic proof backing its Receipt; hence its Receipt provides an offline verifiable proof of tamper evident registration.
```

File: draft-ietf-scitt-architecture.md
Comment (yogeshbdeshpande) [inline draft-ietf-scitt-architecture.md:225]:
```suggestion
A receipt consists of an Transparency Service-specific inclusion proof for the Signed Statement, a signature by the Transparency Service of the state of the Append-only Log after the inclusion, and additional metadata (contained in the signature's protected headers) to assist in auditing.
```

File: draft-ietf-scitt-architecture.md
Comment (robinbryce) [reply draft-ietf-scitt-architecture.md]: I also prefer offline verifiable though

File: draft-ietf-scitt-architecture.md
Comment (robinbryce) [inline draft-ietf-scitt-architecture.md:223]:
```suggestion
Receipts are based on Signed Inclusion Proofs, such as those as described in COSE Signed Merkle Tree Proofs {{-COMETRE}};
```

File: draft-ietf-scitt-architecture.md
Comment (yogeshbdeshpande) [inline draft-ietf-scitt-architecture.md:419]:
```suggestion
The security properties of the Append-only Log are determined by the choice of the verifiable data structure used by the Transparency Service to implement the Log.
```

File: draft-ietf-scitt-architecture.md
Comment (yogeshbdeshpande) [inline draft-ietf-scitt-architecture.md:433]:
```suggestion
: the Append-only Log includes sufficient information to enable authorised actors with access to its content to check that each included Signed Statement has been correctly registered.
```

File: draft-ietf-scitt-architecture.md
Comment (yogeshbdeshpande) [reply draft-ietf-scitt-architecture.md:377]: I agree with @SteveLasker assessment and I would prefer the earlier `old` version of the text.

In fact I would re-phrase it to:
`An Append-only Log is the verifiable data structure that registers Signed Statements and produces Receipts.`

File: draft-ietf-scitt-architecture.md
Comment (SteveLasker) [reply draft-ietf-scitt-architecture.md]: Resolving as I've merged in the _offline verifiable_ change. Please re-open if folks don't agree.

File: draft-ietf-scitt-architecture.md
Comment (robinbryce) [reply draft-ietf-scitt-architecture.md:378]: This gives me something extra to think about, sounds like a generaly useful thing, but doesn't seem necessary architecture ?

Comment (SteveLasker) [reply draft-ietf-scitt-architecture.md:378]: Resolving as not-needed, and implementation specific details.
@ad-l, please re-open with comments and changes if you feel we need to keep it

File: draft-ietf-scitt-architecture.md
Comment (SteveLasker) [reply draft-ietf-scitt-architecture.md]: Marking as resolved: Please see the files view for the most recent changes.

File: draft-ietf-scitt-architecture.md
Comment (SteveLasker) [reply draft-ietf-scitt-architecture.md:224]: While COMETRE is currently the only one defined, the spec allows for more. Not sure the spec needs to list all the current and potential future implementations. 
To get this merged, I'm deferring this to a potential new PR, if felt needed.

File: draft-ietf-scitt-architecture.md
Comment (aj-stein-nist) [inline draft-ietf-scitt-architecture.md:90]: This informative RFC is about trust anchor management and that is material to the changes being mad, did you or someone reviewing intend to remove it?

Sorry suggestions does not support adding back deleted lines anymore or I would just recommend it.

**Review by aj-stein-nist — CHANGES_REQUESTED** [review]

File: draft-ietf-scitt-architecture.md
Comment (yogeshbdeshpande) [reply draft-ietf-scitt-architecture.md:90]: Trust Anchor Management is a vital part of SCITT story. I suggest we should retain it, perhaps needs to be referred from the section that talks about setting the Transparency Service with necessary Trust Anchors!

Comment (SteveLasker) [reply draft-ietf-scitt-architecture.md:90]: Good catch. I removed the reference as it was no longer in the doc, and called out as a make error:

The reference to RFC6024 was removed in [PR 201](https://github.com/ietf-wg-scitt/draft-ietf-scitt-architecture/pull/201/files#diff-585fb424519b79cf000445f3425fa56d328cbaca5d2a622740ddc4e5ca91dbe1L394)

**Review by OR13 — APPROVED** [review]

**Review by SteveLasker — APPROVED** [review]
LGTM

File: draft-ietf-scitt-architecture.md
Comment (JAG-UK) [inline draft-ietf-scitt-architecture.md:225]:
```suggestion
A Receipt consists of a Transparency Service-specific inclusion proof for the Signed Statement, a signature by the Transparency Service of the state of the Append-only Log after the inclusion, and additional metadata (contained in the signature's protected headers) to assist in auditing.
```

**Review by JAG-UK — APPROVED** [review]
Tiny grammar nit

**Review by bobmartin3000 — COMMENTED** [review]
These changes look good

**Pull request merged**
